nucypher-core/Cargo.toml

```diff
@@ -26,3 +26,4 @@ zeroize = { version="1.6.0", features = ["derive"] }
 rand_core = "0.6.4"
 rand_chacha = "0.3.1"
 rand = "0.8.5"
+serde_bytes = "0.11.9"
```

Comment:
I think [`serdect`](https://docs.rs/serdect/latest/serdect/) would be a better choice since it supports human-readable formats too.

Reply:
Could use some guidance here. It doesn't seem as simple as using `serialize_hex_lower_or_bin`/`deserialize_hex_or_bin` (https://docs.rs/serdect/latest/serdect/index.html) for the Serialize/Deserialize impls. I assume I am missing something else...?

Reply:
Yes, you're right, it seems that it's intentional (opened https://github.com/RustCrypto/formats/issues/1111 to clarify). Meanwhile, I think you can use `umbral_pre::serde_bytes` instead of the `serde_bytes` crate - that will give you constant sized output. `Serialize` can be written down explicitly or derived, `Deserialize` has to be written down explicitly since `PublicKey` cannot have the required trait derived for it:

```rust
use umbral_pre::serde_bytes;

/// A session public key.
#[derive(PartialEq, Eq, Hash, Copy, Clone, Debug, Serialize)]
pub struct SessionStaticKey(
    #[serde(serialize_with = "serde_bytes::as_hex::serialize")]
    PublicKey
);

// or explicitly
impl Serialize for SessionStaticKey {
    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
    where
        S: Serializer,
    {
        umbral_pre::serde_bytes::as_hex::serialize(self.0.as_bytes(), serializer)
    }
}

impl serde_bytes::TryFromBytes for SessionStaticKey {
    type Error = core::array::TryFromSliceError;
    fn try_from_bytes(bytes: &[u8]) -> Result<Self, Self::Error> {
        let array: [u8; 32] = bytes.try_into()?;
        Ok(SessionStaticKey(PublicKey::from(array)))
    }
}

impl<'a> Deserialize<'a> for SessionStaticKey {
    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
    where
        D: Deserializer<'a>,
    {
        serde_bytes::as_hex::deserialize(deserializer)
    }
}
```